pkg/pods/pod.go

```diff
@@ -593,6 +596,34 @@ func (pod *Pod) Launchables(manifest Manifest) ([]launch.Launchable, error) {
 	return launchables, nil
 }
 
+func (pod *Pod) SetFinishExec(finishExec []string) {
+	var runAs string
+
+	if manifest, err := pod.CurrentManifest(); err == nil {
+		runAs = manifest.RunAsUser()
+	} else {
+		runAs = string(pod.Id)
+	}
+
+	p2ExecArgs := p2exec.P2ExecArgs{
+		Command: finishExec,
+		User:    runAs,
+		EnvDirs: []string{pod.EnvDir()},
+	}
+
+	pod.FinishExec = append([]string{pod.P2Exec}, p2ExecArgs.CommandLine()...)
```

Comment:
We'll need to write a `finish` for each launchable, it may be sensible to let the launchable's environment also be part of each one's respective exec